lib/src/utils/uri_extension.dart

```
@@ -21,7 +21,79 @@ import 'dart:core';
 import 'package:matrix/src/client.dart';
 
 extension MxcUriExtension on Uri {
-  /// Returns a download Link to this content.
+  /// Transforms this `mxc://` Uri into a `http` resource, which can be used
+  /// to download the content.
+  ///
+  /// Throws an exception if the scheme is not `mxc` or the homeserver is not
+  /// set.
+  ///
+  /// Important! To use this link you have to set a http header like this:
+  /// `headers: {"authorization": "Bearer ${client.accessToken}"}`
+  Future<Uri> getDownloadUri(Client client) async {
+    if (!isScheme('mxc')) throw Exception('Uri has unknown scheme "mxc"');
```

not sure if we should throw a exception or return a empty uri here atm

---

decided not to